### DISCUSSION TITLE
Pending rename: `llm.nvim` -> `model.nvim`

### DISCUSSION THREAD

**Discussion by gsuuon** [gsuuon/llm.nvim#37] [Announcements]
## Pending rename: `llm.nvim` -> `model.nvim`

I'm moving this plugin to `model.nvim` to avoid a name collision with huggingface's plugin since their rename[1]. This means existing users will need to change `require('llm..)` to `require('model..)`. I'll be re-exporting the top-level `require('llm')` with a warning for now, but if you're importing any of the utils, providers, etc (i.e. if you've written your own prompts) you'll need to do a `:%s/require('llm/require('model/`. This comes with a chat feature, command changes and several fixes/improvements.

You can check out the changes early in the [model](https://github.com/gsuuon/llm.nvim/tree/model) branch. The previous version is available as the `llm` tag if you're not ready to switch yet.

### Chat

https://github.com/gsuuon/llm.nvim/assets/6422188/b5082daa-173a-4739-9690-a40ce2c39d15

`:Mchat [name]`

A big value add of using LLM's in Neovim is that it's a great editor for text - the chat experience shouldn't be any different. Chat runs in a normal editable buffer, so it's possible to tune things like the temperature or easily edit an assistant response and continue the conversation. Instead of asking the LLM to fix little simple mistakes, you can just fix the response and continue on. Since it's a normal buffer, you can also just copy-paste the entire conversation (or run `:Mchat [other chat prompt]`) and try again with a different model or settings.

### Commands
Along with the rename, all of the commands have been changed - they'll still work with their old names with a warning.

#### Renamed
`Llm` → `Model` or `M`  
`LlmShow` → `Mshow`  
`LlmSelect` → `Mselect`  
`LlmDelete` → `Mdelete`  
`LlmCancel` → `Mcancel`  

#### Removing
`LlmMulti` -- unchanged and planned to be removed

#### New
`Mchat [name]` - create a new chat buffer with the given chat prompt  
`Mchat` - run the current chat buffer (when `filetype=mchat`)  
`Myank` - yank the selected text with line numbers and filename, useful for dropping file context into a conversation.  
`Mcount` - count the current buffer's tokens using tiktoken (requires python and tiktoken installed)

### Other
- curl data is now passed via stdin instead of as an argument to avoid issues with max argument lengths
- llamacpp server autostart setup and options have been simplified
- llamacpp provider sends content-type header
- added together and ollama providers
- fixed palm and huggingface issues

[1] also maybe I'd like to play around with non-large, non-language models

**Comment by McPatate** [comment]
I want to apologise for not checking if `llm.nvim` existed before making the name change on my plugin 😅

Keep up the good work!

**Reply by gsuuon** [reply]
No problem, it happens 😊. Thanks, and you too!
